plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteJdbcClient.java

```diff
@@ -0,0 +1,562 @@
+     */
+    private static final String IGNITE_SYS_SCHEMA = "SYS";
+    private static final String IGNITE_SCHEMA = "PUBLIC";
+    private final AggregateFunctionRewriter aggregateFunctionRewriter;
+
+    @Inject
+    public IgniteJdbcClient(BaseJdbcConfig config, ConnectionFactory connectionFactory, IdentifierMapping identifierMapping)
+    {
+        super(config, "`", connectionFactory, identifierMapping);
+
+        JdbcTypeHandle bigintTypeHandle = new JdbcTypeHandle(Types.BIGINT, Optional.of("bigint"), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
+        this.aggregateFunctionRewriter = new AggregateFunctionRewriter(
+                this::quoted,
+                ImmutableSet.<AggregateFunctionRule>builder()
+                        .add(new ImplementCountAll(bigintTypeHandle))
+                        .add(new ImplementCount(bigintTypeHandle))
+                        .add(new ImplementMinMax())
+                        .add(new ImplementSum(IgniteJdbcClient::toTypeHandle))
+                        .add(new ImplementAvgFloatingPoint())
+                        .add(new ImplementAvgDecimal())
+                        .add(new ImplementAvgBigint())
+                        .build());
+    }
+
+    @Override
+    public Collection<String> listSchemas(Connection connection)
+    {
+        return ImmutableList.of(IGNITE_SYS_SCHEMA, IGNITE_SCHEMA);
+    }
+
+    @Override
+    public ResultSet getTables(Connection connection, Optional<String> schemaName, Optional<String> tableName)
+            throws SQLException
+    {
+        DatabaseMetaData metadata = connection.getMetaData();
+        return metadata.getTables(connection.getCatalog(),
+                connection.getSchema(),
+                tableName.orElse(null),
+                null);
+    }
+
+    @Override
+    public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        // TODO complete type mapping
+        switch (typeHandle.getJdbcType()) {
+            case Types.VARCHAR:
+                return Optional.of(varcharColumnMapping(
+                        typeHandle.getColumnSize()
+                                // default added column in Ignite will has Integer.MAX_VALUE length which not allow in Trino, will consider the column as varchar
+                                .filter(columnSize -> columnSize != UNBOUNDED_LENGTH)
+                                .map(VarcharType::createVarcharType)
+                                .orElse(createUnboundedVarcharType()), true));
+            case Types.DATE:
+                return Optional.of(ColumnMapping.longMapping(
+                        DATE,
+                        dateReadFunction(),
+                        dateWriteFunction()));
+        }
+        return legacyColumnMapping(session, connection, typeHandle);
```

Comment:
remove. it's fairly easy to add required type mapping (along with the tests)
however, once the connector is merged into master, it's hard to remove `legacyColumnMapping` usage

```suggestion
        return Optional.empty();
```